node/src/Unison/Runtime/ResourcePool.hs

```diff
@@ -36,8 +35,8 @@ addResourceToMap release pool p r wait maxPoolSize getThread poolSizeR queue = d
       >> STM.atomically (TQ.writeTQueue queue key)
     else release r
 
-_acquire :: (Ord p, H.Hashable p) => (p -> IO r) -> (r -> IO ()) -> Cache p r -> MaxPoolSize -> IORef.IORef Int -> ReaperQueue p -> IO CC.ThreadId -> p -> Int -> IO (r, IO ())
-_acquire acquire release pool maxPoolSize poolSizeR queue getThread p wait = do
+_acquire :: (Ord p, H.Hashable p) => (p -> IO r) -> (r -> IO ()) -> Cache p r -> Int -> MaxPoolSize -> IORef.IORef Int -> ReaperQueue p -> IO CC.ThreadId -> p -> IO (r, IO ())
+_acquire acquire release pool wait maxPoolSize poolSizeR queue getThread p = do
   threadId <- getThread
   m <- M.lookup (p,threadId) pool
```

Comment:
@steveshogren Okay, so I'm not sure that keying the map on `(p, ThreadId)` is ideal, sorry I didn't spot this sooner. That means that if a resource is first acquired by thread A, then released, then thread B comes along before expiration and attempts to acquire the same resource, it won't get the cached one. So to get the benefit of using the pool, you need to acquire all your resources on the same thread. I think that is little too fragile.

I get that we need some way of preventing the resource from being simultaneously leased out to two different threads. Can you come up with a solution for that? Might be as simple as:

```Haskell
type IsLeased = Bool
type Cache p r = M.Map p (r, IsLeased, UTCTime, IO ())
```

But I have not thought about it much.

If you feel you are running out of steam on this, I can merge as is and we can address later with a separate PR.